gnosis/cowswap
# Cache issues

# Summary

This PR is an attempt to solve the caching issues of our website

I think the issue has to do with the Service Worker caching the index.html
I followed this guide to allow the worker to update when there's a new version ready:
https://dev.to/atonchev/flawless-and-silent-upgrade-of-the-service-worker-2o95 

## Edit 11/11/21
I have a strong suspicion, it's the worker, and it happens if the user doesn't close the tab. As long as they have one tab open with the old version, they will keep seeing the old version.

This PR aims to solve this, it tries to detect this situation, invalidate some caches.

If it works, 🤞, future PRs could even try to give more info to the users so they refresh, but we need to run more tests on this before deciding what we should do exactly.

# To Test

1. I added some console logs
2. We could try to deploy RCs to staging and see if we manage to get it to automatically udate
